src/Tools/dotnet-monitor/FormatterOptions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel;
+
+#if UNITTEST
+namespace Microsoft.Diagnostics.Monitoring.UnitTests.Options
+#else
+namespace Microsoft.Diagnostics.Tools.Monitor
+#endif
+{
+    internal sealed class FormatterOptions
```

Comment:
The `simple` and `json` formatters have more options beyond these. These are just the common options. You could try using this as a base class, however it probably won't be understood by IntelliSense providers.

I think your other options are to (a) add all of the properties among all of the formatters into the same class and call out in the display strings to which formatters they apply, or (b) create a class for each formatter and use [implication](https://json-schema.org/understanding-json-schema/reference/conditionals.html#implication) to apply the schemas conditionally based on what value of FormatterName is. The latter gives a richer experience where IntelliSense is only offered for the properties that apply to the specified formatter name; it can also provide a default schema if the formatter name is not specified.